docs/design-v1.md

```diff
@@ -0,0 +1,106 @@
+# Caching Service Design Doc
+
+## Purpose
+
+Provide a generalized cache store that can be used from any other service, allowing users to avoid re-running expensive computation that has already been completed for the same dataset.
+
+## MVP Specifications
+
+**_Set a cache value_**
+
+To set a cache value, send a POST request with content-type "multipart/form-data" (to facilitate the file upload).
```

Comment:
Multipart is such a pain. An alternative is doing one post with the file, and another post with the json (or whatever)

Reply:
I agree multipart is a pain, but I'd also argue that two requests is a pain. You can also do a mixed json/form request, but that's also a pain. Was thinking we'd provide a python client, and so most people wouldn't know about the content-type.

Reply:
Two requests is a pain, but IMO it's much less of a pain than multipart. That's because every library has to have good support of plain old requests or no  one is going to use the library. However, multipart is a much rarer operation and so libs often don't support it or don't support it well. Having had to deal with multipart upload a few times, I'd much rather make two standard requests than one multipart request.